vnext/local-cli/config/projectConfig.js

```diff
@@ -1,66 +1,164 @@
+  };
 
-function findSolution(folder) {
-  const solutionPath = glob.sync(path.join('**', '*.sln'), {
-    cwd: folder,
-    ignore: ['node_modules/**', '**/Debug/**', '**/Release/**', 'Generated Files'],
-  })[0];
+  var validProject = false;
 
-  return solutionPath ? path.join(folder, solutionPath) : null;
-}
+  if (usingManualOverride) {
+    // Manual override, try to use it for solutionFile
+    if (!('solutionFile' in userConfig)) {
+      result.solutionFile =
+        'Error: Solution file is required but not specified in react-native.config.';
+    } else if (userConfig.solutionFile === null) {
+      result.solutionFile =
+        'Error: Solution file is null in react-native.config.';
+    } else {
+      result.solutionFile = userConfig.solutionFile;
+    }
 
-function findCppProject(folder, projectName) {
-  const cppProj = glob.sync(path.join('**', projectName + '.vcxproj'), {
-    cwd: folder,
-    ignore: ['node_modules/**', '**/Debug/**', '**/Release/**', '**/Generated Files/**', '**/packages/**'],
-  })[0];
+    // Manual override, try to use it for project
+    if (!('project' in userConfig)) {
+      result.project =
+        'Error: Project is required but not specified in react-native.config.';
+    } else if (userConfig.project === null) {
+      result.project = 'Error: Project is null in react-native.config.';
+    } else {
+      if (!('projectFile' in userConfig.project)) {
+        result.project = {
+          projectFile:
+            'Error: Project file is required for project in react-native.config.',
+        };
+      } else if (userConfig.project.projectFile === null) {
+        result.project = {
+          projectFile: 'Error: Project file is null in react-native.config.',
+        };
+      } else {
+        result.project = {
+          projectFile: userConfig.project.projectFile,
+        };
+        validProject = true;
+      }
+    }
+  } else {
+    // No manually provided solutionFile, try to find it
+    const foundSolutions = configUtils.findSolutionFiles(sourceDir);
+    if (foundSolutions.length === 0) {
+      result.solutionFile =
+        'Error: No app solution file found, please specify in react-native.config.';
+    } else if (foundSolutions.length > 1) {
+      result.solutionFile =
+        'Error: Too many app solution files found, please specify in react-native.config.';
+    } else {
+      result.solutionFile = foundSolutions[0];
+    }
 
-  return cppProj ? path.join(folder, cppProj) : null;
-}
+    // No manually provided project, try to find it
+    const foundProjects = configUtils.findAppProjectFiles(sourceDir);
+    if (foundProjects.length === 0) {
+      result.project = {
+        projectFile:
+          'Error: No app project file found, please specify in react-native.config.',
+      };
+    } else if (foundProjects.length > 1) {
+      result.project = {
+        projectFile:
+          'Error: Too many app project files found, please specify in react-native.config.',
```

Comment:
Do I understand correctly that this requires editing the config file to hint to the CLI which solution to build?
Users should be able to build any project with `npx react-native run-windows` with some arguments if it isn't obvious which solution/project to build. Storing these things in the config adds friction. For example, if we wanted to build both playground and playgroun-win32, do we now need to edit the config file in the pipeline ? is this config file checked in?

Reply:
It looks like this still uses the `--sln`  option so it sounds like there are more than one way which isn't bad but we just need to be careful with documenting this. Can you include a markdown doc pretty please :) ?
The way I understand it the behavior is:
- if user passes `--sln` then use that solution
- otherwise, check `react-native.config` for which solution to use

Then to determine the app project we use the existence of the right `.targets` file `Import` statement which I'd like to make more explicit and instead use a particular property in the project so that it's clear what this does and we aren't tied to the name or existence of the targets import

Reply:
`react-native.config` is the official way to override this for the cli. It's at the core of the how the cli operates, all cli commands are supposed to use the information provided by config. This is why I was saying that our RNW CLI commands are fundamentally not integrating with RN commands, duplicating functionality and making it more confusing for RN developers who expect to modify the config if necessary. I think it's the wrong approach to optimize for the 0.01% case that playground represents.